SLIP-0002: Increase Ledger Wide Read Bytes Limit for Soroban  [DRAFT]

tomerweller opened the pull request.

dmkozh commented on limits/slp-0002.md:
```diff
@@ -0,0 +1,44 @@
+
+## Goals Alignment
+
+Increasing the limit will support existing protocols to operate at acale, faciliating efficient and scalable everyday financial services.
```
typo: scale

dmkozh commented on limits/slp-0002.md:
```diff
@@ -0,0 +1,44 @@
+
+Soroban is often in surge pricing (20%-40% of Ledgers, Early February 2025) and the only resource getting exhausted is the read bytes.
+
+Examining a simple [SoroSwap invocation](https://stellar.expert/explorer/public/tx/238820179026497536#238820179026497537) reveals that 64kb are read, mostly due to reading WASM (two contracts in this instance). With the current ledger limit of 500kb this amounts to a theoretical limit of 7 swaps per ledger. Other ledger wide limits (read entries, write entries, instructions, ledger write bytes) a ...
```
nit: WASM->Wasm

Shaptic commented on limits/slp-0002.md:
```diff
@@ -0,0 +1,44 @@
+
+Soroban is often in surge pricing (20%-40% of Ledgers, Early February 2025) and the only resource getting exhausted is the read bytes.
+
+Examining a simple [SoroSwap invocation](https://stellar.expert/explorer/public/tx/238820179026497536#238820179026497537) reveals that 64kb are read, mostly due to reading WASM (two contracts in this instance). With the current ledger limit of 500kb this amounts to a theoretical limit of 7 swaps per ledger. Other ledger wide limits (read entries, write entries, instructions, ledger write bytes) a ...
```
> mostly due to reading WASM (two contracts in this instance)

isn't this significantly improved after the module cache CAP gets implemented?

tomerweller replied:
Yeah the proposed caching will completely negate this issue. However, We don't have a P23 mainnet date yet and in the mean while we're surge pricing on less than 1.5 simple swaps per second

tomerweller commented:
@dmkozh addressed your comments here: https://github.com/stellar/stellar-protocol/pull/1644

I'm willing to pay for a benchmark, just tell me where to send the XLM.
